### PR TITLE
[GREAT-68] GET List of Group

### PR DESCRIPTION
**Pull request by QuocHuy178**
Testing result: 
image

### CODE REVIEW

File: src/Services/GTT/GTT.Api/Configuration/Routes.cs [hunk 1]
```diff
@@ -32,6 +32,13 @@ public static class Routes
     public const string GetExGroup = "v1/excercise-group";
     #endregion
 
+    /// <summary>
+    /// Group route 
+    /// </summary>
+    #region GroupManagement
+    public const string Group = "v1/group";
```
Comment (ClearActionConsistent) [inline src/Services/GTT/GTT.Api/Configuration/Routes.cs]: the route should be v1/groups

File: src/Services/GTT/GTT.Api/GTT.Api.csproj [hunk 2]
```diff
@@ -7,6 +7,11 @@
     <Nullable>enable</Nullable>
     <RootNamespace>GTT_API</RootNamespace>
   </PropertyGroup>
+  <ItemGroup>
+    <Compile Remove="AzureStorageManagement\**" />
+    <EmbeddedResource Remove="AzureStorageManagement\**" />
```
Comment (ClearActionConsistent) [inline src/Services/GTT/GTT.Api/GTT.Api.csproj]: these line of code is irrelevant

File: src/Services/GTT/GTT.Api/GroupManagement/GetGroups.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            _logger = loggerFactory.CreateLogger<GetGroups>();
+            _mediator = mediator;
+        }
+        #endregion
+
+        [Function("GetGroup")]
+        [OpenApiOperation(nameof(GetGroups), "Groups")]
+        [OpenApiParameter("PageSize", In = ParameterLocation.Query, Required = false, Type = typeof(int))]
+        [OpenApiParameter("PageIndex", In = ParameterLocation.Query, Required = false, Type = typeof(int))]
+        [OpenApiParameter("Keyword", In = ParameterLocation.Query, Required = false, Type = typeof(string))]
```
Comment (ClearActionConsistent) [inline src/Services/GTT/GTT.Api/GroupManagement/GetGroups.cs]: We should create a separated API for SearchGroups, so remove Keywork parameter for this API.

Or 

If we keep this API for SearchGroup, then create new API for GetGroups

File: src/Services/GTT/shared/GTT.Application/Interfaces/Repositories/IGroupRepository.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GTT.Application.Interfaces.Repositories
+{
+    public interface IGroupRepository
+    {
+        Task<ListGroupResponse> GetAllGroup(int pageSize, int pageIndex, string filter);
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Interfaces/Repositories/IGroupRepository.cs]: I think the name GetGroups is more descriptive. as discussed before, we will create a separated API for search. so remove filter param

File: src/Services/GTT/shared/GTT.Application/Queries/GetGroup.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+using MediatR;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GTT.Application.Queries
+{
+    public class GetGroup
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Queries/GetGroup.cs]: change the name to GetGroups, both file name and class name

**Review by phantuananh18 — APPROVED** [review]

**Review by ClearActionConsistent — APPROVED** [review]

**Pull request merged**